src/libs/actions/Report.js

```diff
@@ -1314,6 +1314,12 @@ function updateNotificationPreference(reportID, notificationPreference) {
     API.Report_UpdateNotificationPreference({reportID, notificationPreference});
 }
 
+function navigateToConciergeDMReport() {
+    debugger;
+    const reportID = lodashGet(findConciergeDMReport(allReports), 'reportID');
+    Navigation.navigate(ROUTES.getReportRoute(9));
```

Comment:
You mentioned that this navigation isn't working... So, I have a couple of questions.

1. Is `reportID` correct?
2. Is whatever returned from `ROUTES.getReportRoute(reportID)` correct?
3. Is the route that you're navigating to, different than the route that it's currently on?

If I assume that you've checked all those and the answer is yes to all of them, then I really don't have any idea why it wouldn't work. We use this same navigation technique in three other places in this file, so I am dumbfounded on why it wouldn't work.

Reply:
You could try to move all this code to be in `TriggerWelcomePage.js`, but I don't know if that makes a difference or not. Since the job of an action file is to "manage the data in Onyx", it seems a bit odd to have this code be in the action.

Reply:
1 and 2 are yes and 3 doesn't really matter I think (tried same and different).
Yeah I tried doing `Navigation.navigate(ROUTES.getReportRoute(9))` over [here](https://github.com/Expensify/Expensify.cash/pull/3963/files#diff-d78e8eb2d2b1e86572c5eee906724143f47e00a676c7e1e862dd31236b6987c0R36) and it still fails giving me,
```
The action 'NAVIGATE' with payload {"name":"Report","params":{"reportID":"9"}} was not handled by any navigator.

Do you have a screen named 'Report'?
```
But it works just fine if I call it in the [.then](https://github.com/Expensify/Expensify.cash/pull/3963/files#diff-d78e8eb2d2b1e86572c5eee906724143f47e00a676c7e1e862dd31236b6987c0R38). must be some race condition. Anyway shall change all of this based on our [other](https://github.com/Expensify/Expensify.cash/pull/3963/files#r670102238) discussion.

Reply:
Wow, I've never seen that error before. It sounds like it's trying to navigate to the report page before the report page is mounted, and I am not sure how that's possible. Maybe the report page hasn't been mounted yet because it's data is still being loaded into Onyx? What if you navigate somewhere else instead, like home?